qrand/platforms/cirq/circuit.py

```diff
@@ -19,85 +19,117 @@
+
 
     def rx(self, radians: float, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Rx(radians, self.qubits[target_qubit]))
 
     def ry(self, radians: float, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Ry(radians,self.qubits[target_qubit]))
 
     def rz(self, radians: float, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Rz(radians, self.qubits[target_qubit]))
 
     def s(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.S(self.qubits[target_qubit]))
 
     def t(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
-    def u1(self, theta: float, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
-    def u2(self, phi: float, lam: float, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
-    def u3(
-        self, theta: float, phi: float, lam: float, target_qubit: int
-    ) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.T(self.qubits[target_qubit]))
 
     def x(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.X(self.qubits[target_qubit]))
+
 
     def y(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Y(self.qubits[target_qubit]))
+
 
     def z(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Z(self.qubits[target_qubit]))
 
     ############################# TWO QUBIT GATES #############################
-    def cs(self, control_qubit: int, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
-    def cx(self, control_qubit: int, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
```

Comment:
Official Implementation of CX, CS are WIP. We can wait until the CIRQ ships them officially.